### ISSUE TITLE
[BUG] MaskedDataset initializes y_masked from x instead of y

### ISSUE THREAD

**Issue by blenbot** [gc-os-ai/pyaptamer#599]
**Describe the bug**

` MaskedDataset.__getitem__` initializes y_masked from x instead of y, which can corrupt masked training targets when input and target sequences differ.
**To Reproduce**

```python
import torch
from pyaptamer.datasets.dataclasses import MaskedDataset

# x and y intentionally different
x_data = [[1, 2, 3, 0]]
y_data = [[9, 9, 9, 0]]

dataset = MaskedDataset(
    x=x_data,
    y=y_data,
    max_len=4,
    mask_idx=99,
    masked_rate=0.5,
    is_rna=False,
)

x_masked, y_masked, x, y = dataset[0]

print("x:", x.tolist())
print("y:", y.tolist())
print("y_masked:", y_masked.tolist())
```

**Expected behavior**

```
x: [1, 2, 3, 0]                                                          
y: [9, 9, 9, 0]                                                          
y_masked: [0, 9, 0, 0]
```

**What Actually Happened**
```
x: [1, 2, 3, 0]                                                          
y: [9, 9, 9, 0]                                                          
y_masked: [0, 0, 3, 0] 
```
**Additional context**

**Versions**
<details>

0.1.0a1

</details>

**Comment by pranavchoudhary-tech** [comment]
Hi! I am a prospective applicant for ESoC 2026 and would like to claim this issue as part of my application prerequisite. I have investigated the codebase and already have a fix ready. I will be opening a Pull Request with the changes shortly!

**Comment by blenbot** [comment]
> Hi! I am a prospective applicant for ESoC 2026 and would like to claim this issue as part of my application prerequisite. I have investigated the codebase and already have a fix ready. I will be opening a Pull Request with the changes shortly!

I already added a fix while raising the issue, there is no need for that

**Comment by siddharth7113** [comment]
#304  #356
